packages/data-stores/src/launch/actions.ts

```diff
@@ -124,7 +143,30 @@ export const hideModalTitle = () =>
 		type: 'HIDE_MODAL_TITLE',
 	} as const );
 
-export type LaunchAction = ReturnType<
+export const enablePersistentSuccessView = () =>
+	( {
+		type: 'ENABLE_SUCCESS_VIEW',
+	} as const );
+
+export const disablePersistentSuccessView = () =>
+	( {
+		type: 'DISABLE_SUCCESS_VIEW',
+	} as const );
+
+/**
+ * Usually we use ReturnType of all the action creators to deduce all the actions.
+ * This works until one of the action creators is a generator and doesn't actually "Return" an action.
+ * This type helper allows for actions to be both functions and generators
+ */
+type ReturnOrGeneratorYieldUnion< T extends ( ...args: any ) => any > = T extends (
+	...args: any
+) => infer Return
+	? Return extends Generator< infer T, infer U, any >
+		? T | U
+		: Return
+	: never;
```

Comment:
If this better describes action creators, we should move it with other helpers and use it in other store action creators as well.

https://github.com/Automattic/wp-calypso/blob/61acc957638fc7009632af4c0c058aa6bc0e82b7/packages/data-stores/src/mapped-types.ts#L31-L61

This is the type of thing wp-data should handle in its type system and I think a lot of the utilities we've developed here can move there when the library is typed. This is the kind of thing the system should take care of if we get it right and we shouldn't have to deal with them to build a soundly typed store 🙂